### DISCUSSION TITLE
Want to confirm license status for semi-integration of LSP4IJ into a commercial plugin product

### DISCUSSION THREAD

**Discussion by SCWells72** [redhat-developer/lsp4ij#611] [Q&A] [answered]
Hi. I'm the developer of a commercial JetBrains plugin that supports both commercial and free/Community Edition (CE) JetBrains IDEs. LSP4IJ can help address a major gap for users of my plugin who also use a free/CE IDE, specifically the lack of first-class support for JavaScript/TypeScript/CSS.

I originally thought I'd just document installation and configuration of LSP4IJ for those users, but I actually managed to implement a very tight on-boarding process that helps them to install/enable the plugin and, once installed, to create the respective language server definitions. The latter is based on an optional plugin dependency and only triggers once the user has successfully installed/enabled the LSP4IJ plugin in their IDE instance. While it uses static  compile-time references to a few types from `com.redhat.devtools.lsp4ij.*`, none of the LSP4IJ distribution is bundled with my product.

I just wanted to confirm that this usage is compliant with the governing Eclipse Public License v2.0 that is used for this project and such an integration doesn't force any change to my own commercial product's licensing. I do give proper recognition of LSP4IJ as a Red Hat project and not part of my own product with a link to the LSP4IJ entry in the JetBrains plugin repository.

Please let me know if there are any other considerations. Either way I definitely want to introduce users of my plugin to this wonderful project for the benefits it can provide to them, and I'd prefer it if my own product could help streamline the integration vs. just pointing them at a config doc.

Thanks in advance!

**Comment by fbricon** [comment, marked as answer]
Yes, you're fine as long as you're not creating an LSP4IJ fork.

**Comment by angelozerr** [comment]
I close the discussion @SCWells72 since @fbricon give you an answer
